system/Security/Security.php

```diff
@@ -288,7 +288,11 @@ public function CSRFVerify(RequestInterface $request)
 	 */
 	public function CSRFSetCookie(RequestInterface $request)
 	{
-		$expire        = time() + $this->CSRFExpire;
+		if ($this->CSRFExpire !== 0) {
+			$expire        = time() + $this->CSRFExpire;
+		} else {
+			$expire        = $this->CSRFExpire;
+		}
```

Comment:
```suggestion
		$expire        = $this->CSRFExpire === 0 ? $this->CSRFExpire : time() + $this->CSRFExpire;
```

Reply:
Certainly that is more appropriate. to fix and commit.